services/admin/src/admin/routes/refresh_outdated_cache.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+from typing import Optional
+
+from libcommon.config import ProcessingGraphConfig
+from libcommon.processing_graph import ProcessingStep
+from libcommon.queue import Priority, Queue
+from libcommon.simple_cache import get_cache_info_for_kind_minor_than_version
+from starlette.requests import Request
+from starlette.responses import Response
+
+from admin.authentication import auth_check
+from admin.utils import (
+    AdminCustomError,
+    Endpoint,
+    UnexpectedError,
+    get_json_admin_error_response,
+    get_json_ok_response,
+)
+
+
+def create_refresh_outdated_cache_endpoint(
+    processing_step: ProcessingStep,
+    processing_graph_config: ProcessingGraphConfig,
+    hf_endpoint: str,
+    hf_token: Optional[str] = None,
+    external_auth_url: Optional[str] = None,
+    organization: Optional[str] = None,
+) -> Endpoint:
+    async def refresh_outdated_cache_endpoint(request: Request) -> Response:
+        try:
+            logging.info(f"/refresh-outdated-cache{processing_step.cache_kind}")
+
+            # if auth_check fails, it will raise an exception that will be caught below
+            auth_check(external_auth_url=external_auth_url, request=request, organization=organization)
+
+            current_version = processing_graph_config.specification[processing_step.cache_kind]["version"]
+
+            logging.info(f"refresh cache entries with worker_version<{current_version}")
+            for cache_info in get_cache_info_for_kind_minor_than_version(processing_step.cache_kind, current_version):
+                Queue().upsert_job(
```

Comment:
Not sure if it is enough to upsert a new Job. I saw that the previous outdated cache will remain in db, should we delete it?

Reply:
No, that's it. We want to serve the outdated cache entry until we have a better one. And in parallel, create jobs with priority=LOW to update them (also force=True? not sure, since they will be updated anyway due to the new major job runner's version)

Reply:
If we will keep the outdated versions, the validation should be: Look for cache entries with worker_version minor that the current one BUT without a newest cache entry. Otherwise, it will be creating Jobs with LOW priority in each deployment.

Reply:
I'm not sure to understand. We don't keep the outdated versions, the cache entries will be updated, and the worker_version field with them, so next time, this cache will not be included in the refresh batch